src/linter/html/linter.ts

```diff
@@ -1,32 +1,27 @@
-import {taskStart} from "../../detectors/util/perf.js";
-import {extractJSScriptTags} from "../../detectors/transpilers/html/parser.js";
-import {LintMessageSeverity} from "../../detectors/AbstractDetector.js";
-import HtmlReporter from "./HtmlReporter.js";
+import {LinterParameters} from "../LinterContext.js";
+import transpileHtml from "./transpiler.js";
+import {Resource} from "@ui5/fs";
 
-import type {TranspileResult} from "../../detectors/transpilers/AbstractTranspiler.js";
-import type {ReadStream} from "node:fs";
+export default async function lintHtml({workspace, context}: LinterParameters) {
+	let htmlResources: Resource[];
+	const filePaths = context.getFilePaths();
+	if (filePaths?.length) {
+		htmlResources = [];
+		await Promise.all(filePaths.map(async (filePath) => {
+			if (!filePath.endsWith(".html")) {
+				return;
+			}
+			const resource = await workspace.byPath(filePath);
+			if (!resource) {
+				throw new Error(`Resource not found: ${filePath}`);
+			}
+			htmlResources.push(resource);
+		}));
+	} else {
+		htmlResources = await workspace.byGlob("**/{*.html}");
+	}
 
-export async function lintHtml(resourceName: string, contentStream: ReadStream): Promise<TranspileResult> {
-	const taskLintEnd = taskStart("Linting HTML", resourceName);
-	const report = new HtmlReporter(resourceName);
-	const jsScriptTags = await extractJSScriptTags(contentStream);
-
-	jsScriptTags.forEach((tag) => {
-		const scriptContent = tag.textNodes?.map((tNode) => tNode.value).join("").trim();
-
-		if (scriptContent) {
-			report.addMessage({
-				node: tag,
-				severity: LintMessageSeverity.Warning,
-				ruleId: "ui5-linter-csp-unsafe-inline-script",
-				message: `Use of unsafe inline script`,
-				messageDetails: "{@link topic:fe1a6dba940e479fb7c3bc753f92b28c Content Security Policy}",
-			});
-		}
-	});
-
-	taskLintEnd();
-
-	const {messages} = report.getReport();
-	return {messages, source: "", map: ""};
+	await Promise.all(htmlResources.map(async (resource: Resource) => {
+		await transpileHtml(resource.getPath(), resource.getStream(), context);
+	}));
```

Comment:
Done. I kept the `await Promise.all` so that if an exception in transpileHtml is thrown, the stack trace will include this module (`html/linter.ts`). Without the await, the stack trace would only include the calling module (or whichever ultimately waits for the promise to resolve)